apps/emqx_rule_engine/src/emqx_rule_registry.erl

```diff
@@ -203,7 +216,14 @@ delete_rule(RuleId) when is_binary(RuleId) ->
         {ok, Rule} -> delete_rule(Rule);
         not_found -> ok
     end;
-delete_rule(Rule = #rule{}) when is_record(Rule, rule) ->
+delete_rule(Rule = #rule{id = Id, for = Topics}) when is_record(Rule, rule) ->
```

Comment:
The guard `when is_record(Rule, rule)` is needless?

Reply:
Yes, I'll remove it